src/components/ui/navbar.tsx

```diff
@@ -6,11 +6,22 @@ import { type Variants, motion } from "framer-motion";
 import { HomeIcon, InfoIcon, LaptopIcon, MenuIcon, XIcon } from "lucide-react";
 import Image from "next/image";
 import Link from "next/link";
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { usePathname } from "next/navigation";
 
 const Navbar: React.FC = () => {
   const changeNav = useNavbarScroll();
   const [showSidebar, setShowSidebar] = useState(false);
+  const pathname = usePathname();
+  const initialActivePage =
+    pathname === "/"
+      ? 0
+      : pathname === "/blog"
+        ? 1
+        : pathname === "/tentang-kami"
+          ? 2
+          : 0;
+  const [activePage, setActivePage] = useState(initialActivePage);
```

Comment:
please separate the logic from the page, maybe it can be changed into separate hooks like "useActivePage"